src/main/java/com/google/code/or/common/util/MySQLUtils.java

```diff
@@ -79,6 +79,13 @@ public static int toYear(int value) {
         return new java.sql.Time(millis + (nanosForFractionalValue(fraction, width) / 1000000));
 	}
 
+	public static java.sql.Timestamp toTimestamp2FromRawTime(int value, int fraction, int width) {
+		final java.sql.Time time = toTime2(value, fraction, width);
+		final java.sql.Timestamp t = new java.sql.Timestamp(time.getTime());

```

Comment:
Same as above, it's like `long seconds` and `int value` don't represent the same thing.
I'll give it another look tomorrow.